src/main/java/io/nats/client/api/MessageInfo.java

```diff
@@ -83,12 +127,20 @@ public Headers getHeaders() {
         return headers;
     }
 
+    public String getStream() {
+        return stream;
+    }
+
+    public long getLastSeq() {
+        return lastSeq;
+    }
+
     @Override
     public String toString() {
         return "MessageInfo{" +
                 "subject='" + subject + '\'' +
                 ", seq=" + seq +
-                ", data='" + data + '\'' +
+                ", data='" + (data == null ? "null" : new String(data, UTF_8)) + '\'' +
```

Comment:
also need the rest of the direct header data

Reply:
fixed